csrc/kernel_cache.h

```diff
@@ -228,6 +232,11 @@ class TORCH_CUDA_CU_API FusionKernelRuntime {
   //! Cache of all tensors in the complete fusion
   std::vector<TensorView*> all_tvs_;
 
+  //! store number of arguments in KernelArgumentHolder after each segment
+  //! used to check if arguments are erased if not being used in the following
+  //! segments
+  std::vector<int> args_num_after_segment_runs_;
```

Comment:
nit: `num_live_tensors_after_segment_runs_`

Reply:
it not only holds tensor args, so changed to num_live_args_after_segment_runs_